native/src/main/java/io/ballerina/stdlib/websocket/client/listener/SyncClientConnectorListener.java

```diff
@@ -108,6 +116,22 @@ public void onMessage(WebSocketTextMessage webSocketTextMessage) {
                         callback.complete(WebSocketUtil
                                 .createWebsocketError(String.format("data binding failed: %s", message),
                                         WebSocketConstants.ErrorCode.Error));
+                    } else if (this.targetType != null) {
+                        boolean validationEnabled = connectionInfo.getWebSocketEndpoint().getMapValue(
+                                WebSocketConstants.CLIENT_ENDPOINT_CONFIG).getBooleanValue(ENABLE_VALIDATION_CONFIG);
+                        if (validationEnabled) {
+                            Object validationResult = Constraints.validate(message, this.targetType);
+                            if (validationResult instanceof BError) {
+                                callback.complete(WebSocketUtil.createWebsocketErrorWithCause(
+                                        String.format("data validation failed: %s", validationResult),
+                                        WebSocketConstants.ErrorCode.PayloadValidationError,
+                                        (BError) validationResult));
+                            } else {
+                                callback.complete(message);
+                            }
+                        } else {
+                            callback.complete(message);
```

Comment:
cant we use 1 `callback.complete()` instead of 3

Reply:
We can do it by getting the negation of all the if conditions. But making it that way will reduce the readability and it will become difficult to modify and maintain. Do you have any other suggestions?

Reply:
We can remove #130 and #133 lines and have only 1 `callback.complete()`

Reply:
We can't do it. If we remove those else blocks have only one at the end, it will complete the callback twice when the `validationResult` is a `BError`.